File: java/muse/util/RandomFactoryTest.java
import muse.util.RandomFactory;
import muse.util.TestRunner;

public class RandomFactoryTest {

    public static boolean testGenIntN() {
        int value;
        for (int i = 0; i < 8192; i++) {
            if (RandomFactory.genIntN(0, 0) != 0) {
                return false;
            }

            if (RandomFactory.genIntN(1, 1) != 1) {
                return false;
            }

            value = RandomFactory.genIntN(0, 1);
            if (value < 0 || value > 1) {
                return false;
            }

            value = RandomFactory.genIntN(100, 10000);
            if (RandomFactory.genIntN(value, value) != value) {
                return false;
            }
            if (value < 100 || value > 10000) {
                return false;
            }
        }

        return true;
    }

    public static boolean testGenEven() {
        for (int i = 0; i < 8192; i++) {
            if ((RandomFactory.genEven() & 1) != 0) {
                return false;
            }
        }

        return true;
    }

    public static boolean testGenOdd() {
        for (int i = 0; i < 8192; i++) {
            if ((RandomFactory.genOdd() & 1) == 0) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        TestRunner.pick(RandomFactoryTest::testGenIntN);

        TestRunner.pick(RandomFactoryTest::testGenEven);

        TestRunner.pick(RandomFactoryTest::testGenOdd);
    }
}

File: python/muse/util/random_factory_test.py
from muse.util import random_factory, test_runner


def test_gen_int_n() -> bool:
    value = 0
    for _ in range(8192):
        if random_factory.gen_int_n(0, 0) != 0:
            return False

        if random_factory.gen_int_n(1, 1) != 1:
            return False

        value = random_factory.gen_int_n(0, 1)
        if value < 0 or value > 1:
            return False

        value = random_factory.gen_int_n(100, 10000)
        if random_factory.gen_int_n(value, value) != value:
            return False
        if value < 100 or value > 10000:
            return False

    return True


def test_gen_even() -> bool:
    for _ in range(8192):
        if (random_factory.gen_even() & 1) != 0:
            return False

    return True


def test_gen_odd() -> bool:
    for _ in range(8192):
        if (random_factory.gen_odd() & 1) == 0:
            return False

    return True


def main() -> None:
    test_runner.pick(test_gen_int_n)

    test_runner.pick(test_gen_even)

    test_runner.pick(test_gen_odd)


if __name__ == "__main__":
    main()
